### ISSUE TITLE
Kata-containers snap installed... now what?

### ISSUE THREAD

**Issue by pnunn** [kata-containers/documentation#201]
I know this is going to sound lame, but, I've installed the snap on Ubuntu and now have no clue how to get a container running. 

I've had a look at the docs.. hmmm.. I've run lxd, I've run kvm with libvirt, but I have no idea how to start one of these (and they sound exactly what I need). Do I install docker and go from there? Is that the approach? Any simple "how to's" available that I havn't been able to find?

P.

**Comment by grahamwhaley** [comment]
Short answer - you need to use either Docker or Kubernetes.

Slightly longer answer :-)...
Hi @pnunn - I guess you followed the instructions to install the snap over at https://github.com/kata-containers/packaging/tree/master/snap - yes?
I believe that snap gets you the 'kata containers OCI runtime' installed. That runtime can then be used with Docker or Kubernetes - your choice. See the section half way down that previous link (at https://github.com/kata-containers/packaging/tree/master/snap#integration-with-docker-and-kubernetes) which will then direct you to how to configure Docker or Kubernetes to use that snap installed runtime.

Hope that helps.
/cc @devimc

**Comment by pnunn** [comment]
Hmm... OK, no I didn't do that. I used the software store to install the snap. Perhaps I need to do a bit more :) I did look at where it has installed but the code seems to already exist without any sign of a make file.  Is that correct or do I need to do the manual install?  Thanks for the help. I've also managed to install Docker and got hello-world working so looking seriously a this.

**Comment by elimisteve** [comment]
Similarly to op: `README.md` has shown me how to install kata containers, but how do I actually spin up a container?  _Must_ we change the default Docker runtime to kata containers, then use Docker as we normally would?  It looks like we should be able to do

```
docker run --runtime=kata-runtime -it alpine /bin/bash
```

but I get the following error, even after restarting Docker:

```
docker: Error response from daemon: Unknown runtime specified kata-container.
See 'docker run --help'.
```

Any idea why this doesn't work?  I'd like to be able to use `--runtime` to experiment with kata container and not switch everything over to kata-runtime right away.

Please document how to do this!

**Comment by jodh-intel** [comment]
Hi @elimisteve - You don't need to change the *default* Docker runtime, but you do need to "register" `kata-runtime` with Docker by adding it as an available runtime as shown in step 2 in:

- https://github.com/kata-containers/documentation/blob/master/install/docker/ubuntu-docker-install.md

To see which runtimes Docker has available:

```sh
$ sudo docker info | grep -i runtime
Runtimes: runc kata-runtime
Default Runtime: runc
```

However, I do think the [current snap instructions](https://github.com/kata-containers/documentation/blob/master/install/snap-installation-guide.md) are ... cough... rather impressively terse :)

@devimc - please could you use this issue to make these docs clearer? What I think we need to do:

- Modify the `snap-installation-guide.md` so that it more closely matches the format in the other (non-snap) installation guides (with the link to the docker and k8s pages).

- Since snap isn't necessarily Ubuntu-specific, I think we need a new `install/docker/snap-docker-install.md` that specifies the correct path to the `kata-runtime` binary as `/snap/kata-containers/current/usr/bin/kata-runtime`.

  Basically, move as much of https://github.com/kata-containers/packaging/tree/master/snap into the user docs as possible (and make the packaging docs reference the user docs).

**Comment by jodh-intel** [comment]
(Bullets above updated to reflect the fact there is no point creating `install/docker/snap-docker-install.md` as those instructions are by definition distro-specific, and since snap is not tied to Ubuntu, that is not the best approach).

**Comment by GabyCT** [comment]
Closing this issue because a guide of how to use snap and install snap is provided at https://github.com/kata-containers/packaging/tree/master/snap#integration-with-docker-and-kubernetes
